test/BoardTests.cpp

```diff
@@ -70,10 +70,147 @@ TEST_F(BoardTests, checkBoardArgumentBoard_WithGetBoardLinesFunction)
+    ASSERT_EQ(board.getBoardLines().at(2).at(1), BoardFields::WHITE);
+}
+
+TEST_F(BoardTests, checkBoardGetField)
+{
+    Board board(3, 2, {}, {});
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(0, 1, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getField(1, 1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getField(1, 0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getField(0, 1), BoardFields::WHITE);
+    ASSERT_EQ(board.getField(0, 0), BoardFields::BLACK);
+}
+
+TEST_F(BoardTests, checkBoardSetFieldException)
+{
+    Board board(2, 3, {}, {});
+    ASSERT_THROW(board.setField(2, 0, BoardFields::BLACK), std::out_of_range);
+    ASSERT_THROW(board.setField(0, 3, BoardFields::BLACK), std::out_of_range);
+}
+
+TEST_F(BoardTests, checkBoardGetFieldException)
+{
+    Board board(2, 3, {}, {});
+    ASSERT_THROW(board.getField(2, 0), std::out_of_range);
+    ASSERT_THROW(board.getField(0, 3), std::out_of_range);
+}
+
+TEST_F(BoardTests, checkBoardGetRow)
+{
+    Board board(3, 2, {}, {});
+    BLine result1 = {BoardFields::BLACK, BoardFields::UNKNOWN};
+    BLine result2 = {BoardFields::WHITE, BoardFields::UNKNOWN};
+
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(1, 0, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getRow(0), result1);
+    ASSERT_EQ(board.getRow(1), result2);
+}
+
+TEST_F(BoardTests, checkBoardSetRow)
+{
+    Board board(3, 2, {}, {});
+    BLine result = {BoardFields::UNKNOWN, BoardFields::WHITE};
+
+    board.setRow(1, result);
+    ASSERT_EQ(board.getRow(1), result);
+}
+
+TEST_F(BoardTests, checkBoardGetRowException)
+{
+    Board board(2, 4, {}, {});
+    ASSERT_THROW(board.getRow(3), std::out_of_range);
+}
+
+TEST_F(BoardTests, checkBoardSetRowException)
+{
+    Board board(2, 4, {}, {});
+    BLine lineTooBig = {BoardFields::UNKNOWN, BoardFields::WHITE, BoardFields::UNKNOWN};
+    BLine lineTooShort = {BoardFields::UNKNOWN};
+    BLine lineGood = {BoardFields::UNKNOWN, BoardFields::WHITE};
+
+    ASSERT_THROW(board.setRow(1, lineTooBig), std::out_of_range);
+    ASSERT_THROW(board.setRow(1, lineTooShort), std::out_of_range);
+    ASSERT_THROW(board.setRow(3, lineGood), std::out_of_range);
+}
+
+TEST_F(BoardTests, checkBoardGetCol)
+{
+    Board board(2, 3, {}, {});
+    BLine result1 = {BoardFields::BLACK, BoardFields::UNKNOWN};
+    BLine result2 = {BoardFields::UNKNOWN, BoardFields::UNKNOWN};
+    BLine result3 = {BoardFields::UNKNOWN, BoardFields::WHITE};
+
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(1, 2, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getCol(0), result1);
+    ASSERT_EQ(board.getCol(1), result2);
+    ASSERT_EQ(board.getCol(2), result3);
+}
+
+TEST_F(BoardTests, checkBoardSetCol)
+{
+    Board board(2, 3, {}, {});
+    BLine result = {BoardFields::UNKNOWN, BoardFields::WHITE};
+    BLine untouched = {BoardFields::UNKNOWN, BoardFields::UNKNOWN};
+
+    board.setCol(1, result);
+    ASSERT_EQ(board.getCol(0), untouched);
```

Comment:
getCol is already tested in above test. The same for other functions like getRow and getField. You can have both operations in one test. That's how you should usually test setters + getters.

Reply:
done